APE9.rst

```diff
@@ -0,0 +1,111 @@
+A General Timeseries Class for Astropy
+-------------------------------------
+
+author: Stuart Mumford
+
+date-created: 2016 March 24
+
+date-last-revised: 2016 March 24
+
+type: Standard Track
+
+status: Discussion
+
+
+Abstract
+--------
+
+The goal of a timeseries object in astropy is to provide a core set of
+functionality that can be shared between astropy, affiliated packages and the
+wider community.
+
+The scope of this timeseries object is designed to remain focused on the core
+functionality needed, and not to try and replace or re-implement the
+functionality in the pandas library. The main motivation for this class over
+using pandas is primarily to support the functionality of other parts of the
+astropy package such as Time, Units and Coordinates packages.
+
+This document proposes an implementation based on Table, with extra constraints
+and specific functionality dedicated to timeseries type data.
+
+Detailed description
+--------------------
+
+Many different areas of astrophysics have to deal with 1D timeseries data, i.e.
+either sampling a continuous variable at fixed times or counting some events
+binned into time windows. These types of applications require some basic
+functionality:
+
+#. Extending timeseries with extra rows
+#. Concatenating multiple timeseries objects
+#. Sorting
+#. Slicing / selecting time ranges (indexing)
+#. Re-binning and re-sampling timeseries
+#. Interpolating to different time stamps.
+#. Masking
+#. Support for subtraction and addition.
+#. Plotting and visualisation.
+
+While this functionality is found in non-domain specific packages such as
+pandas, a timeseries object in Astropy would also provide some functionality
+which is more related to Astronomy:
+
+#. Converting between time systems
+#. Astropy unit support
+#. Support variable width time bins.
+
+It is proposed that this functionality be implemented via a subclass of the
+``QTable`` class, with a few requirements specific to the timeseries class:
+
+#. A 'Time' index column exists which enforces unique indexes.
+#. The table is always sorted in terms of increasing time.
```

Comment:
it needs to be able to sort itself, the list above is not end-user functionality.